server/graphql/v2/object/Expense.ts

```diff
@@ -546,14 +546,46 @@ export const GraphQLExpense = new GraphQLObjectType<ExpenseModel, express.Reques
           if (expense.status === expenseStatus.DRAFT) {
             let draftFields = EXPENSE_DRAFT_PUBLIC_FIELDS;
             let itemsFields = EXPENSE_DRAFT_ITEMS_PUBLIC_FIELDS;
-            if (await ExpenseLib.canSeeExpenseDraftPrivateDetails(req, expense)) {
+            const canSeeExpenseDraftPrivateDetails = await ExpenseLib.canSeeExpenseDraftPrivateDetails(req, expense);
+            if (canSeeExpenseDraftPrivateDetails) {
               draftFields = [...draftFields, ...EXPENSE_DRAFT_PRIVATE_FIELDS];
               itemsFields = [...itemsFields, ...EXPENSE_DRAFT_ITEMS_PRIVATE_FIELDS];
             }
 
             const draftData = pick(expense.data, draftFields);
-            if (expense.data?.items) {
-              draftData.items = (expense.data.items as any[]).map(item => pick(item, itemsFields));
+            const items = ((expense.data.items as { url?: string }[]) || []).map(item => pick(item, itemsFields));
+            for (const item of items) {
+              if (item.url) {
+                const uploadedFile = await req.loaders.UploadedFile.byUrl.load(item.url);
```

Comment:
:ice_cream: Because we're in a for loop, the loader here will actually be slower than a simple `model.find` (there's no batching possible, yet you get the loaders overload). A more performant way to do that would have been to use a `Promise.all`, which is compatible with loaders.